core/src/main/scala/org/apache/spark/deploy/worker/ui/WorkerPage.scala

```diff
@@ -33,7 +33,12 @@ import org.apache.spark.ui.{UIUtils, WebUIPage}
 import org.apache.spark.util.Utils
 
 private[ui] class WorkerPage(parent: WorkerWebUI) extends WebUIPage("") {
-  private val workerEndpoint = parent.worker.self
+  private val workerEndpoint = {
```

Comment:
Same here: if we don't feel comfortable adding infinite loop, we can just change it from val to def, with adding comment it shouldn't be cached.